File: csharp/Conjured.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace csharp
{
    internal class Conjured : Item
    {
        public Conjured(string name, int sellIn, int quality) : base(name, sellIn, quality)
        {
            TypeOfItem = ItemType.Conjured;
        }

        protected override void QualityModifier()
        {
            if (this.SellIn >= 0) SetQuality(2);
            else SetQuality(4);
        }
    }
}

File: python/src/conjured.py
from python.components.gilded_rose import GildedRose, Item




class ConjuredItem(GildedRose):

    def __init__(self, name, quality, sell_in):
        super().__init__(Item(name, quality, sell_in))

    def update_quality(self):
        if self.quality > 0:
            self.quality -= 2
        if self.sell_in <= 0 and self.quality > 0:
            self.quality -= 2
        self.sell_in -= 1
        return self.quality, self.sell_in